frontends/web/src/new_front/components/CreateSamples/CreateSamples/AnnotationInterfaces/Contexts/ChatWithInstructions.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { FC, useContext, useEffect, useState } from "react";
+import SignContract from "new_front/components/Modals/SignContract";
+import SimpleChatbot from "new_front/components/CreateSamples/CreateSamples/AnnotationInterfaces/Contexts/SimpleChatbot";
+import useFetch from "use-http";
+import { ContextAnnotationFactoryType } from "new_front/types/createSamples/createSamples/annotationFactory";
+import { ContextConfigType } from "new_front/types/createSamples/createSamples/annotationContext";
+import UserContext from "containers/UserContext";
+import Modal from "react-bootstrap/Modal";
+import { ChatHistoryType } from "new_front/types/createSamples/createSamples/utils";
+import { CreateInterfaceContext } from "new_front/context/CreateInterface/Context";
+
+const ChatWithInstructions: FC<
+  ContextAnnotationFactoryType & ContextConfigType
+> = ({ taskId, generative_context, setIsGenerativeContext, contextId }) => {
+  const [signInConsent, setSignInConsent] = useState(false);
+  const { post, response } = useFetch();
+  const [chatHistory, setChatHistory] = useState<ChatHistoryType>({
+    user: [],
+    bot: [],
+  });
+  const [modelName, setModelName] = useState("");
+  const [provider, setProvider] = useState("");
+  const [artifactsInput, setArtifactsInput] = useState<any>(
+    generative_context.artifacts,
+  );
+  const [finishConversation, setFinishConversation] = useState(false);
+  const { updateModelInputs } = useContext(CreateInterfaceContext);
+  const { user } = useContext(UserContext);
+
+  const checkIfUserIsSignedInConsent = async () => {
+    const signConsent = await post("/task/check_sign_consent", {
```

Comment:
We should rename the endpoint we're calling to check_signed_consent. Not urgent, but definitely an improvement.